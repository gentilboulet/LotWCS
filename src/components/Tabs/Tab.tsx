import * as React from 'react';

export interface ITabProps {
    tabId: string;
    title: string;
}

class Tab extends React.Component<ITabProps, object> {}

export { Tab };
