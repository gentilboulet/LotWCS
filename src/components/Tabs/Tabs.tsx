import * as classnames from 'classnames';
import * as React from 'react';
import { Container, Nav, NavItem, NavLink, TabContent, TabPane } from 'reactstrap';

export interface ITabsProps {
  defaultTab?: string;
}

interface ITabsState {
  activeTab?: string;
}

class Tabs extends React.Component<ITabsProps, ITabsState> {
  constructor(props: ITabsProps) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      activeTab: this.props.defaultTab,
    };
  }
  public render() {
    /* tslint:disable:no-any */
    return (
      <Container>
        <Nav tabs={true}>
        {React.Children.map(this.props.children, (child: any) => {
          return (<NavItem>
                    <NavLink
                      className={classnames({ active: this.state.activeTab === child.props.tabId })}
// tslint:disable-next-line jsx-no-lambda
                      onClick={() => this.toggle(child.props.tabId)}
                    >
                      {child.props.title}
                    </NavLink>
                  </NavItem>);
          }
        )}
        </Nav>
        <TabContent activeTab={this.state.activeTab}>
        {React.Children.map(this.props.children, (child: any) => {
          return (<TabPane tabId={child.props.tabId} >
                    {child.props.children}
                  </TabPane>);
          }
        )}
        </TabContent>
      </Container>
    );
  }

  private toggle(tab: string) {
    if (this.state.activeTab !== tab) {
      this.setState({
        activeTab: tab,
      });
    }
  }
}

export { Tabs };
