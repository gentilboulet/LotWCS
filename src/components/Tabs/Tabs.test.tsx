import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';

import { Tab } from './Tab';
import { Tabs } from './Tabs';

describe('Testing the Tabs component', () => {  
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Tabs />, div);
  });

  it('should render the inner Tabs', () => {
    const tree = renderer.create(
      <Tabs defaultTab="first">
        <Tab title="2nd" tabId="second"> The second content </Tab>
        <Tab title="1st" tabId="first"> The first content </Tab>
      </Tabs>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
