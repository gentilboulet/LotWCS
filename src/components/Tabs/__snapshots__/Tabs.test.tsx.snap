// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Tabs component should render the inner Tabs 1`] = `
<div>
  <ul
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        onClick={[Function]}
      >
        2nd
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="active nav-link"
        onClick={[Function]}
      >
        1st
      </a>
    </li>
  </ul>
  <div
    className="tab-content"
  >
    <div
      className="tab-pane"
    >
       The second content 
    </div>
    <div
      className="tab-pane active"
    >
       The first content 
    </div>
  </div>
</div>
`;
